### PR TITLE
NIAD-1296: GitHub bugs

### PR DESCRIPTION
**Pull request by bartek-sarul**
## Description

Please explain the changes you have made here.

## Jira Ticket

Please add a link to the related Jira ticket (if any).

## Checklist

These are items (excluding GitHub Checks) which should be confirmed before a branch is ready to merge.

- [ ] Acceptance Criteria met
- [ ] Commit messages are meaningful
- [ ] Manually tested
- [ ] Self-reviewed your code
- [ ] Code reviewed from two other developers
- [ ] If your pull request depends on any other, please link them in the description
- [ ] main branch is passing/green on CI
- [ ] Add/update any relevant documentation

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -121,6 +121,24 @@ Screening examples can be found [here](https://github.com/nhsconnect/integration
...
+- `Practitioner.identifier.system` - field will be omitted.
+- `ProcedureRequest.reasonReference` - field becomes optional.
+- `Specimen.accessionIdentifier` - field becomes optional.
+- `Specimen.collection` - field becomes optional.
+- `Specimen.identifier` - EDIFACT has nothing to map from. FHIR field will be omitted.
+- `Specimen.type.coding` - EDIFACT does not give Read2 nor SNOMED. `coding.system` will be omitted. Only `coding.display` will be populated.
+- `Observation.code.coding.code` - field becomes optional.
+- `Observation.code.coding.display` - field becomes optional.
+- `Observation.identifier` - EDIFACT has nothing to map from. FHIR field will be omitted.
+- `Observation.value[x]` - can be not only `valueQuantity` but also `valueCodeableConcept`.
```
Comment (mattd-kainos) [inline README.md:140]: Had to read this one a few times, simply it?

```suggestion
- `Observation.value[x]` - can be `valueQuantity` or `valueCodeableConcept`.
```

Comment (matthewdoreilly) [inline README.md:140]: Can `Observation.value[x]` have both a `valueQuantity` and `valueCodeableConcept`?
If so, you could reword this to be "can be `valueQuantity` and/or `valueCodeableConcept`"

Comment (matthewdoreilly) [inline README.md:140]: Can `Observation.value[x]` have both a `valueQuantity` and `valueCodeableConcept`?
If so, you could reword this to be "can be `valueQuantity` and/or `valueCodeableConcept`"

File: src/intTest/resources/fhir/multi_pathology_msg1.fhir.json [hunk 2]
```diff
@@ -1,23 +1,23 @@
 {
     "resourceType": "Bundle",
-    "id": "800ff80a-1bac-4648-840c-1a7f530ac017",
+    "id": "f6afe7cd-090a-4b97-99cb-088e12f821a6",
     "meta": {
-        "lastUpdated": "2021-06-07T10:18:28.432+01:00",
+        "lastUpdated": "2021-06-08T10:47:52.871+01:00",
```
Comment (matthewdoreilly) [inline src/intTest/resources/fhir/multi_pathology_msg1.fhir.json:5]: I think we may have to create a task ticket to have these kind of timestamps a fixed date. Imagine it's annoying having to update them to the current timestamp each time the integration tests are ran

**Review by mattd-kainos — DISMISSED** [review]

File: src/test/java/uk/nhs/digital/nhsconnect/lab/results/translator/mapper/ObservationMapperTest.java [hunk 3]
```diff
@@ -484,7 +484,7 @@ void testMapTestGroupsWithBidirectionalReferences() {
         final var uuidToUrlMap = Map.of("uuid-1", "url-1", "uuid-2", "url-2");
         when(uuidGenerator.generateUUID()).thenReturn("uuid-1", "uuid-2");
         uuidToUrlMap.forEach((uuid, url) ->
-            when(fullUrlGenerator.generate(uuid)).thenReturn(url));
+            lenient().when(fullUrlGenerator.generate(uuid)).thenReturn(url));
```
Comment (matthewdoreilly) [inline src/test/java/uk/nhs/digital/nhsconnect/lab/results/translator/mapper/ObservationMapperTest.java:487]: Why this change? Is this line even still needed?

File: src/intTest/resources/fhir/multi_pathology_msg1.fhir.json (hunk 2 above)
Comment (bartek-sarul) [reply src/intTest/resources/fhir/multi_pathology_msg1.fhir.json:5]: It's not that I have to update it but I copy paste the result replacing whole file. That's why it got updated here

File: src/test/java/uk/nhs/digital/nhsconnect/lab/results/translator/mapper/ObservationMapperTest.java (hunk 3 above)
Comment (bartek-sarul) [reply src/test/java/uk/nhs/digital/nhsconnect/lab/results/translator/mapper/ObservationMapperTest.java:487]: It is needed but only for some value. I just didn't want to spend more time on figuring this out

**Review by mattd-kainos — APPROVED** [review]

File: src/test/java/uk/nhs/digital/nhsconnect/lab/results/translator/mapper/ObservationMapperTest.java (hunk 3 above)
Comment (matthewdoreilly) [reply src/test/java/uk/nhs/digital/nhsconnect/lab/results/translator/mapper/ObservationMapperTest.java:487]: Ah I see, in some cases it's hit and some cases it's not

**Review by matthewdoreilly — APPROVED** [review]
